com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationServiceHandler.java

```diff
@@ -0,0 +1,43 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.Map;
+
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.CloudServiceBinding;
+
+import com.sap.cloud.lm.sl.cf.process.util.ApplicationAttributeUpdater.UpdateState;
+
+public abstract class ApplicationServiceHandler {
```

Comment:
In general I think that the pattern makes the code more complex than it previously was. :( All classes that extend this one, have constructors with a lot of arguments and some of the arguments are a bit scary (like the consumers). This makes it hard to understand when something would be executed and with what parameters.

Reply:
Deleted class